internal/app/networkd/pkg/reg/reg.go

```diff
@@ -149,3 +151,49 @@ func toCIDR(family uint8, prefix net.IP, prefixLen int) string {
 
 	return ipNet.String()
 }
+
+// Check implements the Health api and provides visibilty into the state of networkd.
+// Ready signifies the daemon (api) is healthy and ready to serve requests.
+func (r *Registrator) Check(ctx context.Context, in *empty.Empty) (reply *healthapi.HealthCheckResponse, err error) {
+	reply = &healthapi.HealthCheckResponse{
+		Messages: []*healthapi.HealthCheck{
+			{
+				Status: healthapi.HealthCheck_SERVING,
+			},
+		},
+	}
+
+	return reply, nil
+}
+
+// Watch implements the Health api and provides visibilty into the state of networkd.
+// Ready signifies the daemon (api) is healthy and ready to serve requests.
+func (r *Registrator) Watch(in *empty.Empty, srv healthapi.Health_WatchServer) (err error) {
+	return srv.Send(&healthapi.HealthCheckResponse{
+		Messages: []*healthapi.HealthCheck{
+			{
+				Status: healthapi.HealthCheck_SERVING,
+			},
+		},
+	})
+}
```

Comment:
I like the idea of watching this. Not sure if its just due to the way networkd works or its just not implemented, but I'm curious about the vision for the real world use case for watching.

Reply:
i don't know what the actual health check for networkd looks like (yet?).

Reply:
I think the simplest and most effective health check for now would be a simple ping to the default gateway. There's a reference implementation of a pinger here https://godoc.org/golang.org/x/net/icmp#example-PacketConn--NonPrivilegedPing

```
Check() => ping default gateway
Watch(int) => ping default gateway with frequency in seconds
```

Reply:
Yeah that sounds reasonable. Would be a good indicator that things are working as expected.